lib/ui/auth/signup_screen.dart

```diff
@@ -0,0 +1,307 @@
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+import '../core/themes/deco_theme_extension.dart';
+import '../core/widgets/deco_card.dart';
+import '../core/widgets/deco_primary_button.dart';
+import '../core/widgets/deco_text_field.dart';
+import 'bear_in_love_illustration.dart';
+
+class SignupScreen extends StatefulWidget {
+  const SignupScreen({super.key});
+
+  @override
+  State<SignupScreen> createState() => _SignupScreenState();
+}
+
+class _SignupScreenState extends State<SignupScreen> {
+  final _nicknameCtrl = TextEditingController();
+  final _emailCtrl = TextEditingController();
+  final _pwCtrl = TextEditingController();
+  final _pw2Ctrl = TextEditingController();
```

Comment:
현재 같은 상황에서는 위처럼 '필드 초기화'를 할 때는 문제가 발생하지 않습니다.

하지만 인자값으로 this(자기 자신)를 건네거나, 기본값을 부모로부터 받아서 초기화하거나, 리스너를 사용하려고 한다면 현재 구조보다 initState에 정의하는 것이 좋습니다.
왜냐하면 필드에서 초기화를 하면, 객체가 완전히 생성되기 전에 초기화가 되기 때문에 클래스 내부의 다른 것들에 접근할 수 없습니다.

```
// 오류 발생: 초기화 시점에는 'widget'이나 'this'를 참조할 수 없음
final _emailCtrl = TextEditingController(text: widget.userEmail);
```

```suggestion
// late final을 사용하면 나중에 widget.email 같은 값을 넣기도 편하고 성능상 유리합니다.
  late final TextEditingController _nicknameCtrl;
  late final TextEditingController _emailCtrl;
  late final TextEditingController _pwCtrl;
  late final TextEditingController _pw2Ctrl;
  
  @override
  void initState() {
    super.initState();
    // 여기서 초기화를 몰아서 하면 코드의 흐름(생성 -> 초기화 -> 소멸)이 명확해집니다.
    _nicknameCtrl = TextEditingController();
    _emailCtrl = TextEditingController();
    _pwCtrl = TextEditingController();
    _pw2Ctrl = TextEditingController();
  }
```